File: frontend/src/administrator/creditCard/service/creditCardService.js
import http from '@/config/axios';

export default class CreditCardService {
    constructor() {
        this.endpoint = '/api/v1/credit-cards';
    }
    save(creditCard) {
        if (creditCard.id === null) {
            return http.post(this.endpoint, creditCard).then((response) => response);
        }

        return this.#update(creditCard);
    }

    #update(creditCard) {
        return http.put(this.endpoint, creditCard).then((response) => response);
    }

    findById(id) {}

    findByFilter(filter) {
        return http.get(`${this.endpoint}/filter${filter.toString()}`).then((response) => response);
    }
}

File: backend/credit-card/customer/api/src/main/java/br/com/systec/fintrack/creditcard/service/CreditCardService.java
package br.com.systec.fintrack.creditcard.service;

import br.com.systec.fintrack.commons.exception.BaseException;
import br.com.systec.fintrack.creditcard.commons.CreditCardTransactionType;
import br.com.systec.fintrack.creditcard.filter.CreditCardFilterVO;
import br.com.systec.fintrack.creditcard.model.CreditCard;
import org.springframework.data.domain.Page;

import java.util.List;

public interface CreditCardService {
    CreditCard save(CreditCard creditCard) throws BaseException;
    CreditCard update(CreditCard creditCard, Long id) throws BaseException;
    List<CreditCard> findAll() throws BaseException;
    Page<CreditCard> findByFilter(CreditCardFilterVO filterVO) throws BaseException;
    CreditCard findById(Long id) throws BaseException;
    void updateAvailableLimitCreditCard(double amount, Long id, CreditCardTransactionType transactionType) throws BaseException;
}
